examples/codegen/data_types.rs

```diff
@@ -77,7 +78,7 @@ pub struct FooMessage<'a> {
     pub f_sint32: Option<i32>,
     pub f_sint64: i64,
     pub f_bool: bool,
-    pub f_FooEnum: Option<FooEnum>,
+    pub f_FooEnum: Option<data_types::FooEnum>,
```

Comment:
Let's hang for a few days to get this right. I've started doing a refactoring where I pass a reference to `FileDescriptor` everywhere (rather than separately passing messages and enums) and yes, then enough information is available. It also expresses the dependency on the enclosing file descriptor explicitly.  And let me sort out the missing files so everything merges cleanly!